crates/services/producer/src/block_producer.rs

```diff
@@ -194,6 +194,11 @@ where
                 .expect("It is impossible to overflow the current block height")
         });
 
+        let are_txs_script = transactions
+            .iter()
+            .map(|tx| tx.is_script())
+            .collect::<Vec<bool>>();
```

Comment:
You don't need to `collect` since you just call `iter` on it the next time you use it. Should improve the performance more.

Reply:
Hmm. I just realized that this collect is essentially the same as a `clone` as both allocates to the heap. I'm p sure leaving it as an `iter` solves that problem :crossed_fingers:

Sorry to tie you down with these nits lol.

Reply:
no worries I appreciate the feedback!

Reply:
hmm I don't think we can accomplish this without the `collect` as if we leave it as an `iter` it still points to the original `transactions` vec and eventually gets borrowed later in the if statement

Reply:
Sorry. Yeah. I was worried about that. I think your original clone is better than the collect. It's easier to read.